pysph/cpy/parallel.py

```diff
@@ -507,7 +507,7 @@ def __call__(self, *args, **kw):
 
 class Elementwise(object):
     def __init__(self, func, backend='cython'):
-        if hasattr(func, '__annotations__') and getattr(func, '__annotations__'):
+        if hasattr(func, 'has_annotations'):
```

Comment:
Instead of doing this, you can do this:
```
getattr(func, '__annotations__', None)
```
This avoids having to set another attribute and rely on that.